src/shared/sleep-config.c

```diff
@@ -256,8 +306,36 @@ static bool enough_memory_for_hibernation(void) {
         return r;
 }
 
+int is_product_listed(char **products) {
+        char **product;
+        int r;
+        _cleanup_free_ char *p = NULL;
+
+        if (strv_isempty(products))
+                return false;
+
+        if (access("/sys/class/dmi/id/product_name", W_OK) < 0)
+                return false;
+
+        r = read_one_line_file("/sys/class/dmi/id/product_name", &p);
+        if (r < 0)
+                return false;
+
+        STRV_FOREACH(product, products) {
+                size_t l, k;
+
+                l = strlen(p);
+                k = strlen(*product);
+                if (l == k && memcmp(p, *product, l) == 0)
```

Comment:
Good question, that would be my first approach too. However, I saw this idiom to be a common one in this file (see can_sleep_state() and can_sleep()) and decided to follow it too, assuming I would probably be overlooking something if I just used strcmp().

Sorry, it's a dumb answer, I know. Let's agree on "for consistency"? :)